src/dc_receive_imf.rs

```diff
@@ -344,7 +326,6 @@ fn add_parts(
         } else {
             MessageState::InFresh
         };
-        *to_id = DC_CONTACT_ID_SELF;
```

Comment:
this setting was very weird -- why would we set to_id to ourselves for incoming messages? 
this `to_id` is not much used -- it's the `to_ids` that we use mostly, so probably nothing bad happened.

Reply:
the `to_id` is mainly used to decide later, from reading the database, if a given message is incoming (to_id=SELF) or outgoing (to_id!=SELF).

so i would keep setting the `to_id` explicitly in the incoming/outgoing if-else-branches and rely on the incoming flag - which is set by `from_ids.contains(&DC_CONTACT_ID_SELF)` instead of checking to_ids here - eg. we will have different states if the _first_ to is not self otherwise.

at some point, we can think over if to_id makes sense at all, but for now, i would leave it as is to avoid side-effects.